<a href="https://colab.research.google.com/github/itsGab/previsao_uti_em_covid/blob/main/notebooks/tratamento/papc_tratamento.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<h1 align='center'>Previsão de Admissão por COVID-19</h1>
<h2 align='center'>Tratamento e Limpeza dos Dados</h2>

# Leitura inicial dos dados


In [51]:
data = "https://github.com/itsGab/previsao_uti_em_covid/blob/main/dados/Kaggle_Sirio_Libanes_ICU_Prediction.xlsx?raw=true"

In [52]:
import pandas as pd
df_inicial = pd.read_excel(data)
df_inicial.head()

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65 AGE_PERCENTIL  GENDER  \
0                         0            1          60th       0   
1                         0            1          60th       0   
2                         0            1          60th       0   
3                         0            1          60th       0   
4                         0            1          60th       0   

   DISEASE GROUPING 1  DISEASE GROUPING 2  DISEASE GROUPING 3  \
0                 0.0                 0.0                 0.0   
1                 0.0                 0.0                 0.0   
2                 0.0                 0.0                 0.0   
3                 0.0                 0.0                 0.0   
4                 0.0                 0.0                 0.0   

   DISEASE GROUPING 4  DISEASE GROUPING 5  DISEASE GROUPING 6  HTN  \
0                 0.0                 1.0                 1.0  0.0   
1                 0.0                 1.0                 1.0  0.0   
2                 0.0                 1.0                 1.0  0.0   
3                 0.0                 1.0                 1.0  0.0   
4                 0.0                 1.0                 1.0  0.0   

   IMMUNOCOMPROMISED  OTHER  ALBUMIN_MEDIAN  ALBUMIN_MEAN  ALBUMIN_MIN  \
0                0.0    1.0             NaN           NaN          NaN   
1                0.0    1.0             NaN           NaN          NaN   
2                0.0    1.0        0.605263      0.605263     0.605263   
3                0.0    1.0             NaN           NaN          NaN   
4                0.0    1.0        0.000000      0.000000     0.000000   

   ALBUMIN_MAX  ALBUMIN_DIFF  BE_ARTERIAL_MEDIAN  BE_ARTERIAL_MEAN  \
0          NaN           NaN                 NaN               NaN   
1          NaN           NaN                 NaN               NaN   
2     0.605263          -1.0           -1.000000         -1.000000   
3          NaN           NaN                 NaN               NaN   
4     0.000000          -1.0           -0.871658         -0.871658   

   BE_ARTERIAL_MIN  BE_ARTERIAL_MAX  BE_ARTERIAL_DIFF  BE_VENOUS_MEDIAN  \
0              NaN              NaN               NaN               NaN   
1              NaN              NaN               NaN               NaN   
2        -1.000000        -1.000000              -1.0         -1.000000   
3              NaN              NaN               NaN               NaN   
4        -0.871658        -0.871658              -1.0         -0.863874   

   BE_VENOUS_MEAN  BE_VENOUS_MIN  BE_VENOUS_MAX  BE_VENOUS_DIFF  \
0             NaN            NaN            NaN             NaN   
1             NaN            NaN            NaN             NaN   
2       -1.000000      -1.000000      -1.000000            -1.0   
3             NaN            NaN            NaN             NaN   
4       -0.863874      -0.863874      -0.863874            -1.0   

   BIC_ARTERIAL_MEDIAN  BIC_ARTERIAL_MEAN  BIC_ARTERIAL_MIN  BIC_ARTERIAL_MAX  \
0                  NaN                NaN               NaN               NaN   
1                  NaN                NaN               NaN               NaN   
2            -0.317073          -0.317073         -0.317073         -0.317073   
3                  NaN                NaN               NaN               NaN   
4            -0.317073          -0.317073         -0.317073         -0.317073   

   BIC_ARTERIAL_DIFF  BIC_VENOUS_MEDIAN  BIC_VENOUS_MEAN  BIC_VENOUS_MIN  \
0                NaN                NaN              NaN             NaN   
1                NaN                NaN              NaN             NaN   
2               -1.0          -0.317073        -0.317073       -0.317073   
3                NaN                NaN              NaN             NaN   
4               -1.0          -0.414634        -0.414634       -0.414634   

   BIC_VENOUS_MAX  BIC_VENOUS_DIFF  BILLIRUBIN_MEDIAN  BILLIRUBIN_MEAN  \
0             NaN              NaN                NaN   



Separação de colunas por tipo do dados:

| nome da variável | descrição | 
|---|---|
| cols_comorbidades | grupos de doenças |
| cols_demograficos | faixa etária e sexo |
| cols_sinais_vitais| sinais vitais `iloc[:, 193:-2]`|
| cols_exames_lab | exames laboratoriais `iloc[:, 13:193]`| 
| cols_outros | identificador, janela e uti |

In [53]:
# baseado em um notebook feito pelo criador do desafio
# https://www.kaggle.com/felipeveiga/starter-covid-19-sirio-libanes-icu-admission

# separando em listas os diferentes tipos de colunas

cols_comorbidades = [i for i in df_inicial.columns if "DISEASE" in i]
cols_comorbidades.extend(["HTN", "IMMUNOCOMPROMISED", "OTHER"])

cols_demograficas = [i for i in df_inicial.columns if "AGE_" in i]
cols_demograficas.append("GENDER")

cols_sinais_vitais = df_inicial.iloc[:,193:-2].columns.tolist()

cols_exames_laboratoriais = df_inicial.iloc[:,13:193].columns.tolist()

cols_outras = ["PATIENT_VISIT_IDENTIFIER", "WINDOW", "ICU"]

In [54]:
# Mostrando informações do DataFrame
df_inicial.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1925 entries, 0 to 1924
Columns: 231 entries, PATIENT_VISIT_IDENTIFIER to ICU
dtypes: float64(225), int64(4), object(2)
memory usage: 3.4+ MB


In [55]:
# Numero de PATIENT_VISIT_IDENTIFIER unicos (ou numero de pacientes unicos)
df_inicial['PATIENT_VISIT_IDENTIFIER'].nunique()

385

Informações:

| Info | Descrição |
|--|--|
| Nº de colunas | 231 |
| Nº de linhas | 1925 | 
| Data types| ["float64(225)", "int64(4)", "object(2)"] |
| Nº de pacientes | 385 |  

In [56]:
pd.set_option('display.max_columns', None)
df_inicial.describe()

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65       GENDER  DISEASE GROUPING 1  \
count               1925.000000  1925.000000  1925.000000         1920.000000   
mean                 192.000000     0.467532     0.368831            0.108333   
std                  111.168431     0.499074     0.482613            0.310882   
min                    0.000000     0.000000     0.000000            0.000000   
25%                   96.000000     0.000000     0.000000            0.000000   
50%                  192.000000     0.000000     0.000000            0.000000   
75%                  288.000000     1.000000     1.000000            0.000000   
max                  384.000000     1.000000     1.000000            1.000000   

       DISEASE GROUPING 2  DISEASE GROUPING 3  DISEASE GROUPING 4  \
count         1920.000000         1920.000000         1920.000000   
mean             0.028125            0.097917            0.019792   
std              0.165373            0.297279            0.139320   
min              0.000000            0.000000            0.000000   
25%              0.000000            0.000000            0.000000   
50%              0.000000            0.000000            0.000000   
75%              0.000000            0.000000            0.000000   
max              1.000000            1.000000            1.000000   

       DISEASE GROUPING 5  DISEASE GROUPING 6          HTN  IMMUNOCOMPROMISED  \
count         1920.000000         1920.000000  1920.000000        1920.000000   
mean             0.128125            0.046875     0.213021           0.158333   
std              0.334316            0.211426     0.409549           0.365148   
min              0.000000            0.000000     0.000000           0.000000   
25%              0.000000            0.000000     0.000000           0.000000   
50%              0.000000            0.000000     0.000000           0.000000   
75%              0.000000            0.000000     0.000000           0.000000   
max              1.000000            1.000000     1.000000           1.000000   

             OTHER  ALBUMIN_MEDIAN  ALBUMIN_MEAN  ALBUMIN_MIN  ALBUMIN_MAX  \
count  1920.000000      821.000000    821.000000   821.000000   821.000000   
mean      0.809896        0.528527      0.528527     0.528527     0.528527   
std       0.392485        0.224100      0.224100     0.224100     0.224100   
min       0.000000       -1.000000     -1.000000    -1.000000    -1.000000   
25%       1.000000        0.605263      0.605263     0.605263     0.605263   
50%       1.000000        0.605263      0.605263     0.605263     0.605263   
75%       1.000000        0.605263      0.605263     0.605263     0.605263   
max       1.000000        1.000000      1.000000     1.000000     1.000000   

       ALBUMIN_DIFF  BE_ARTERIAL_MEDIAN  BE_ARTERIAL_MEAN  BE_ARTERIAL_MIN  \
count         821.0          821.000000        821.000000       821.000000   
mean           -1.0           -0.963433         -0.963433        -0.963433   
std             0.0            0.160870          0.160870         0.160870   
min            -1.0           -1.000000         -1.000000        -1.000000   
25%            -1.0           -1.000000         -1.000000        -1.000000   
50%            -1.0           -1.000000         -1.000000        -1.000000   
75%            -1.0           -1.000000         -1.000000        -1.000000   
max            -1.0            1.000000          1.000000         1.000000   

       BE_ARTERIAL_MAX  BE_ARTERIAL_DIFF  BE_VENOUS_MEDIAN  BE_VENOUS_MEAN  \
count       821.000000             821.0        821.000000      821.000000   
mean         -0.963433              -1.0         -0.931121       -0.931121   
std           0.160870               0.0          0.169509        0.169509   
min          -1.000000              -1.0         -1.000000       -1.000000   
25%          -1.000000              -1.0         -1.000000       -1.000000   
50%          -1.000000              -1.0         -1

# Tratamento


## Nome das colunas

In [57]:
#Substituindo espaço por underscore
df_inicial.columns = df_inicial.columns.str.replace(' ', '_')

In [58]:
df_inicial.head()

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65 AGE_PERCENTIL  GENDER  \
0                         0            1          60th       0   
1                         0            1          60th       0   
2                         0            1          60th       0   
3                         0            1          60th       0   
4                         0            1          60th       0   

   DISEASE_GROUPING_1  DISEASE_GROUPING_2  DISEASE_GROUPING_3  \
0                 0.0                 0.0                 0.0   
1                 0.0                 0.0                 0.0   
2                 0.0                 0.0                 0.0   
3                 0.0                 0.0                 0.0   
4                 0.0                 0.0                 0.0   

   DISEASE_GROUPING_4  DISEASE_GROUPING_5  DISEASE_GROUPING_6  HTN  \
0                 0.0                 1.0                 1.0  0.0   
1                 0.0                 1.0                 1.0  0.0   
2                 0.0                 1.0                 1.0  0.0   
3                 0.0                 1.0                 1.0  0.0   
4                 0.0                 1.0                 1.0  0.0   

   IMMUNOCOMPROMISED  OTHER  ALBUMIN_MEDIAN  ALBUMIN_MEAN  ALBUMIN_MIN  \
0                0.0    1.0             NaN           NaN          NaN   
1                0.0    1.0             NaN           NaN          NaN   
2                0.0    1.0        0.605263      0.605263     0.605263   
3                0.0    1.0             NaN           NaN          NaN   
4                0.0    1.0        0.000000      0.000000     0.000000   

   ALBUMIN_MAX  ALBUMIN_DIFF  BE_ARTERIAL_MEDIAN  BE_ARTERIAL_MEAN  \
0          NaN           NaN                 NaN               NaN   
1          NaN           NaN                 NaN               NaN   
2     0.605263          -1.0           -1.000000         -1.000000   
3          NaN           NaN                 NaN               NaN   
4     0.000000          -1.0           -0.871658         -0.871658   

   BE_ARTERIAL_MIN  BE_ARTERIAL_MAX  BE_ARTERIAL_DIFF  BE_VENOUS_MEDIAN  \
0              NaN              NaN               NaN               NaN   
1              NaN              NaN               NaN               NaN   
2        -1.000000        -1.000000              -1.0         -1.000000   
3              NaN              NaN               NaN               NaN   
4        -0.871658        -0.871658              -1.0         -0.863874   

   BE_VENOUS_MEAN  BE_VENOUS_MIN  BE_VENOUS_MAX  BE_VENOUS_DIFF  \
0             NaN            NaN            NaN             NaN   
1             NaN            NaN            NaN             NaN   
2       -1.000000      -1.000000      -1.000000            -1.0   
3             NaN            NaN            NaN             NaN   
4       -0.863874      -0.863874      -0.863874            -1.0   

   BIC_ARTERIAL_MEDIAN  BIC_ARTERIAL_MEAN  BIC_ARTERIAL_MIN  BIC_ARTERIAL_MAX  \
0                  NaN                NaN               NaN               NaN   
1                  NaN                NaN               NaN               NaN   
2            -0.317073          -0.317073         -0.317073         -0.317073   
3                  NaN                NaN               NaN               NaN   
4            -0.317073          -0.317073         -0.317073         -0.317073   

   BIC_ARTERIAL_DIFF  BIC_VENOUS_MEDIAN  BIC_VENOUS_MEAN  BIC_VENOUS_MIN  \
0                NaN                NaN              NaN             NaN   
1                NaN                NaN              NaN             NaN   
2               -1.0          -0.317073        -0.317073       -0.317073   
3                NaN                NaN              NaN             NaN   
4               -1.0          -0.414634        -0.414634       -0.414634   

   BIC_VENOUS_MAX  BIC_VENOUS_DIFF  BILLIRUBIN_MEDIAN  BILLIRUBIN_MEAN  \
0             NaN              NaN                NaN   

## Tratamento de colunas vazias (ou NaN)

### Conceitos e Ação:

1. Cuidado para NÃO usar os dados quando a variável de alvo (‘ICU’ = 1) estiver presente, pois a ordem do evento é desconhecida (talvez o evento de destino tenha acontecido antes de os resultados serem obtidos). Eles foram mantidos lá para que possamos aumentar este conjunto de dados em outros resultados posteriormente.

        Ação: Limpeza dos Pacientes que foram pra UTI na primeira janela.


2. A identificação precoce dos pacientes que desenvolverão um curso adverso da doença (e precisam de cuidados intensivos) é a chave para um tratamento adequado (salvar vidas) e para gerenciar leitos e recursos. O modelo usando apenas o primeiro (0-2) provavelmente será mais clinicamente relevante.

        Ação: Preparar a primeira janela (0-2). Marcando a variável alvo de pacientes que precisam da UTI.

3. Um dos maiores desafios de trabalhar com dados de saúde é que a taxa de amostragem varia entre os diferentes tipos de medições. Por exemplo, os sinais vitais são coletados com mais frequência (geralmente de hora em hora) do que os laboratórios de sangue (geralmente diariamente). É sensato supor que um paciente que não tem uma medição registrada em uma janela de tempo esteja clinicamente estável, podendo apresentar sinais vitais e exames de sangue semelhantes às janelas vizinhas. Portanto, pode-se preencher os valores ausentes usando a entrada seguinte ou anterior.

        Ação: Preencher valores faltantes com backfill, desconsiderando casos que já tenham entrado na UTI.

4. Atenção aos problemas de multicolinearidade e variância zero nesses dados ao escolher seu algoritmo.

        Ação: Remover as colunas com alta correlação

In [59]:
df_tratamento = df_inicial

In [60]:
# listando os pacientes que entraram na uti na primeira janela (0 a 2 horas).
lista_pacientes_uti_primeira_janela = df_tratamento.query('WINDOW == "0-2" & ICU == 1').index.tolist()

In [61]:
# mantendo apenas os pacientes que não estão na lista de entrada na primeira janela.
df_tratamento = df_tratamento.query('PATIENT_VISIT_IDENTIFIER not in @lista_pacientes_uti_primeira_janela')

In [62]:
# definindo função para preparar primeira janela com a marcador da variável alvo.
def preparando_janela(grupo):
    if any(grupo['ICU']):
        grupo.loc[grupo['WINDOW'] == '0-2', 'ICU'] = 1
    return grupo

df_tratamento = df_tratamento.groupby('PATIENT_VISIT_IDENTIFIER').apply(preparando_janela)

In [63]:
# preparo para o preenchimento

# remover as janelas que sejam diferentes das primeira janela e estajam tiveram entrada na UTI
index_linhas_remover = df_tratamento.query('WINDOW != "0-2" & ICU == 1').index.tolist()

In [64]:
df_tratamento = df_tratamento.drop(index=index_linhas_remover)

In [65]:
df_tratamento.query('WINDOW == "0-2"').dropna().describe()

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65     GENDER  DISEASE_GROUPING_1  \
count                 42.000000    42.000000  42.000000           42.000000   
mean                 210.238095     0.547619   0.404762            0.166667   
std                  107.192284     0.503761   0.496796            0.377195   
min                   10.000000     0.000000   0.000000            0.000000   
25%                  136.000000     0.000000   0.000000            0.000000   
50%                  216.500000     1.000000   0.000000            0.000000   
75%                  287.500000     1.000000   1.000000            0.000000   
max                  378.000000     1.000000   1.000000            1.000000   

       DISEASE_GROUPING_2  DISEASE_GROUPING_3  DISEASE_GROUPING_4  \
count           42.000000           42.000000           42.000000   
mean             0.071429            0.095238            0.023810   
std              0.260661            0.297102            0.154303   
min              0.000000            0.000000            0.000000   
25%              0.000000            0.000000            0.000000   
50%              0.000000            0.000000            0.000000   
75%              0.000000            0.000000            0.000000   
max              1.000000            1.000000            1.000000   

       DISEASE_GROUPING_5  DISEASE_GROUPING_6        HTN  IMMUNOCOMPROMISED  \
count           42.000000           42.000000  42.000000          42.000000   
mean             0.190476            0.023810   0.238095           0.190476   
std              0.397437            0.154303   0.431081           0.397437   
min              0.000000            0.000000   0.000000           0.000000   
25%              0.000000            0.000000   0.000000           0.000000   
50%              0.000000            0.000000   0.000000           0.000000   
75%              0.000000            0.000000   0.000000           0.000000   
max              1.000000            1.000000   1.000000           1.000000   

           OTHER  ALBUMIN_MEDIAN  ALBUMIN_MEAN  ALBUMIN_MIN  ALBUMIN_MAX  \
count  42.000000       42.000000     42.000000    42.000000    42.000000   
mean    0.904762        0.595865      0.595865     0.595865     0.595865   
std     0.297102        0.060909      0.060909     0.060909     0.060909   
min     0.000000        0.210526      0.210526     0.210526     0.210526   
25%     1.000000        0.605263      0.605263     0.605263     0.605263   
50%     1.000000        0.605263      0.605263     0.605263     0.605263   
75%     1.000000        0.605263      0.605263     0.605263     0.605263   
max     1.000000        0.605263      0.605263     0.605263     0.605263   

       ALBUMIN_DIFF  BE_ARTERIAL_MEDIAN  BE_ARTERIAL_MEAN  BE_ARTERIAL_MIN  \
count          42.0                42.0              42.0             42.0   
mean           -1.0                -1.0              -1.0             -1.0   
std             0.0                 0.0               0.0              0.0   
min            -1.0                -1.0              -1.0             -1.0   
25%            -1.0                -1.0              -1.0             -1.0   
50%            -1.0                -1.0              -1.0             -1.0   
75%            -1.0                -1.0              -1.0             -1.0   
max            -1.0                -1.0              -1.0             -1.0   

       BE_ARTERIAL_MAX  BE_ARTERIAL_DIFF  BE_VENOUS_MEDIAN  BE_VENOUS_MEAN  \
count             42.0              42.0         42.000000       42.000000   
mean              -1.0              -1.0         -0.970581       -0.970581   
std                0.0               0.0          0.079013        0.079013   
min               -1.0              -1.0         -1.000000       -1.000000   
25%               -1.0              -1.0         -1.000000       -1.000000   
50%               -1.0              -1.0         -1.000000       -1.000000   
75%               -1.0     

In [66]:
df_backfill = df_tratamento.copy()
df_backfill.update(df_backfill.groupby('PATIENT_VISIT_IDENTIFIER', as_index=False).fillna(method='bfill', axis=0))
df_backfill = df_backfill.query('WINDOW == "0-2"')
df_backfill = df_backfill.dropna()

In [67]:
df_backfill.describe()

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65      GENDER  DISEASE_GROUPING_1  \
count                296.000000   296.000000  296.000000          296.000000   
mean                 196.442568     0.415541    0.385135            0.094595   
std                  111.088650     0.493650    0.487451            0.293150   
min                    0.000000     0.000000    0.000000            0.000000   
25%                  104.750000     0.000000    0.000000            0.000000   
50%                  196.500000     0.000000    0.000000            0.000000   
75%                  294.250000     1.000000    1.000000            0.000000   
max                  384.000000     1.000000    1.000000            1.000000   

       DISEASE_GROUPING_2  DISEASE_GROUPING_3  DISEASE_GROUPING_4  \
count          296.000000          296.000000          296.000000   
mean             0.023649            0.081081            0.020270   
std              0.152209            0.273422            0.141162   
min              0.000000            0.000000            0.000000   
25%              0.000000            0.000000            0.000000   
50%              0.000000            0.000000            0.000000   
75%              0.000000            0.000000            0.000000   
max              1.000000            1.000000            1.000000   

       DISEASE_GROUPING_5  DISEASE_GROUPING_6         HTN  IMMUNOCOMPROMISED  \
count          296.000000          296.000000  296.000000         296.000000   
mean             0.114865            0.047297    0.168919           0.162162   
std              0.319399            0.212634    0.375315           0.369224   
min              0.000000            0.000000    0.000000           0.000000   
25%              0.000000            0.000000    0.000000           0.000000   
50%              0.000000            0.000000    0.000000           0.000000   
75%              0.000000            0.000000    0.000000           0.000000   
max              1.000000            1.000000    1.000000           1.000000   

            OTHER  ALBUMIN_MEDIAN  ALBUMIN_MEAN  ALBUMIN_MIN  ALBUMIN_MAX  \
count  296.000000      296.000000    296.000000   296.000000   296.000000   
mean     0.807432        0.583482      0.583482     0.583482     0.583482   
std      0.394984        0.121247      0.121247     0.121247     0.121247   
min      0.000000       -0.578947     -0.578947    -0.578947    -0.578947   
25%      1.000000        0.605263      0.605263     0.605263     0.605263   
50%      1.000000        0.605263      0.605263     0.605263     0.605263   
75%      1.000000        0.605263      0.605263     0.605263     0.605263   
max      1.000000        1.000000      1.000000     1.000000     1.000000   

       ALBUMIN_DIFF  BE_ARTERIAL_MEDIAN  BE_ARTERIAL_MEAN  BE_ARTERIAL_MIN  \
count         296.0          296.000000        296.000000       296.000000   
mean           -1.0           -0.999133         -0.999133        -0.999133   
std             0.0            0.012671          0.012671         0.012671   
min            -1.0           -1.000000         -1.000000        -1.000000   
25%            -1.0           -1.000000         -1.000000        -1.000000   
50%            -1.0           -1.000000         -1.000000        -1.000000   
75%            -1.0           -1.000000         -1.000000        -1.000000   
max            -1.0           -0.786096         -0.786096        -0.786096   

       BE_ARTERIAL_MAX  BE_ARTERIAL_DIFF  BE_VENOUS_MEDIAN  BE_VENOUS_MEAN  \
count       296.000000             296.0        296.000000      296.000000   
mean         -0.999133              -1.0         -0.966570       -0.966570   
std           0.012671               0.0          0.098531        0.098531   
min          -1.000000              -1.0         -1.000000       -1.000000   
25%          -1.000000              -1.0         -1.000000       -1.000000   
50%          -1.000000              -1.0         -1.000000       -1.000000   


In [68]:
p1 =  42/385
p2 = 296/385
p1, p2

(0.10909090909090909, 0.7688311688311689)

Aqui é possível notar que, caso não seja feito o preenchimento dos dados NaN com o metodo "backfill", ou seja, preencher com o próximo valor válido, na hora de descartar as colunas que ainda tenham algum NaN ficariamos com apenas 42 válidos de um total de 385. Já com todo o tratamento e o uso do "backfill" foi possível ficar com 296 pacientes válidos.

|Método de Tratamento| % de Pacientes Válidos|
|--|--|
|Sem Backfill| 10.9 % |
|Com Backfill| 76.9 %|

Depois de todo o preenchimento e tratamento, restaram 296 pacientes

## Trabalhar correlação entre colunas para remover colunas com alta correlação

## Transformar em dados categoricos o "AGE_PERCENTIL"

# Salvar o dataframe em CSV no Repositório do Github

# DELETAR



In [69]:
from sklearn.model_selection import train_test_split
from sklearn.dummy import DummyClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score
from sklearn.metrics import precision_score, precision_recall_fscore_support

In [70]:
df_backfill = df_backfill.drop('WINDOW', axis=1)

In [71]:
X = df_backfill.drop(['ICU', 'AGE_PERCENTIL'], axis=1)
y = df_backfill['ICU']

In [72]:
X_train, X_test, y_train, y_test = train_test_split(X, y)

In [73]:
def roda_modelo(modelo, X, y):
    X_train, X_test, y_train, y_test = train_test_split(X, y)
    mod = modelo().fit(X_train, y_train)
    y_pred = mod.predict(X_test)
    print(f'accuracy:\n> {accuracy_score(y_test, y_pred)}')
    print(f'precision:\n> {precision_score(y_test, y_pred)}')
    print(f'precision:\n> {precision_recall_fscore_support(y_test, y_pred)}')


In [74]:
roda_modelo(DummyClassifier, X, y)

accuracy:
> 0.5405405405405406
precision:
> 0.25925925925925924
precision:
> (array([0.70212766, 0.25925926]), array([0.62264151, 0.33333333]), array([0.66      , 0.29166667]), array([53, 21]))


/usr/local/lib/python3.7/dist-packages/sklearn/dummy.py:132: FutureWarning: The default value of strategy will change from stratified to prior in 0.24.
  "stratified to prior in 0.24.", FutureWarning)


In [75]:
roda_modelo(DecisionTreeClassifier, X, y)

accuracy:
> 0.7027027027027027
precision:
> 0.6086956521739131
precision:
> (array([0.74509804, 0.60869565]), array([0.80851064, 0.51851852]), array([0.7755102, 0.56     ]), array([47, 27]))


In [76]:
roda_modelo(RandomForestClassifier, X, y)

accuracy:
> 0.7432432432432432
precision:
> 0.5882352941176471
precision:
> (array([0.78947368, 0.58823529]), array([0.86538462, 0.45454545]), array([0.82568807, 0.51282051]), array([52, 22]))
